configure && make fails on Mac

**Describe the bug**
Cloning latest master on MacOS Catalina, and then running `configure && make`, as per [instructions here](http://portaudio.com/docs/v19-doxydocs/compile_mac_coreaudio.html), errors out before starting the build. 

Is this still the correct way to build the latest code on Mac?

**To Reproduce**
Steps to reproduce the behavior. Include code if applicable.
1. Clone master branch
2. Run configure && make

**Expected behavior**
The library to be built.

**Actual behavior**
During running of `configure`. a number of errors like the following get output:

```
configure: creating ./config.status
./config.status: line 733: print: command not found
./config.status: line 733: print: command not found
./config.status: line 733: print: command not found
```

During running of `make`:

```
mkdir lib
mkdir -p src/common src/hostapi/alsa src/hostapi/asihpi src/hostapi/asio src/hostapi/coreaudio src/hostapi/dsound src/hostapi/jack src/hostapi/oss src/hostapi/skeleton src/hostapi/wasapi src/hostapi/wdmks src/hostapi/wmme src/os/unix src/os/win
touch lib-stamp
/bin/sh ./libtool --mode=compile gcc -c -std=c99 -O2 -Wall -pedantic -pipe -fPIC -DNDEBUG -DPA_LITTLE_ENDIAN -I./include -I./src/common -I./src/os/unix -Wno-deprecated -Werror -arch x86_64 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -mmacosx-version-min=10.6  -DPACKAGE_NAME=\"\" -DPACKAGE_TARNAME=\"\" -DPACKAGE_VERSION=\"\" -DPACKAGE_STRING=\"\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DSIZEOF_SHORT=2 -DSIZEOF_INT=4 -DSIZEOF_LONG=8 -DHAVE_CLOCK_GETTIME=1 -DHAVE_NANOSLEEP=1 -DPA_USE_COREAUDIO=1 src/common/pa_allocation.c -o src/common/pa_allocation.lo
./libtool: line 170: top_srcdir: command not found
./libtool: line 170: top_srcdir: command not found
./libtool: line 170: top_srcdir: command not found
./libtool: line 338: func_echo_all: command not found
./libtool: line 1037: print: command not found
./libtool: line 1037: print: command not found
./libtool: line 1037: print: command not found
./libtool: line 1037: print: command not found
./libtool: line 1037: print: command not found
./libtool: line 1037: print: command not found
./libtool: line 1037: print: command not found
./libtool: line 900: print: command not found
./libtool: line 900: print: command not found
./libtool: line 971: print: command not found
./libtool: line 973: print: command not found
./libtool: line 907: print: command not found
make: *** [src/common/pa_allocation.lo] Error 1
```

**Desktop (please complete the following information):**
 - OS: MacOS
 - OS Version: 10.15.6 Catalina
 - PortAudio version: master (73291bb1f53ed0a78ef5bd04526e281378244bde)
 - If Windows or Linux, which Host API (e.g. WASAPI): N/A

**Additional info**
I suspect I'm missing some prerequisites on my system but I have Xcode installed and am able to compile C/C++ code from other projects.

Hi @philburk, here are my results.

Going back to commit 68e1c56 does indeed fix the problem. Everything configures and builds fine in both zsh and bash.

I also tried latest master in bash just now as well, and there I don't get the `config.status` and `libtool` errors. The build fails during compilation however with an "Unsupported architecture" error.

```
libtool: compile:  gcc -c -std=c99 -O2 -Wall -pedantic -pipe -fPIC -DNDEBUG -DPA_LITTLE_ENDIAN -I./include -I./src/common -I./src/os/unix -Wno-deprecated -Werror -arch x86_64 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -mmacosx-version-min=10.6 -DPACKAGE_NAME=\"\" -DPACKAGE_TARNAME=\"\" -DPACKAGE_VERSION=\"\" -DPACKAGE_STRING=\"\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DSIZEOF_SHORT=2 -DSIZEOF_INT=4 -DSIZEOF_LONG=8 -DHAVE_CLOCK_GETTIME=1 -DHAVE_NANOSLEEP=1 -DPA_USE_COREAUDIO=1 src/common/pa_cpuload.c  -fno-common -DPIC -o src/common/.libs/pa_cpuload.o
In file included from src/common/pa_cpuload.c:55:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/include/assert.h:42:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/include/sys/cdefs.h:807:2: error: 
      Unsupported architecture
#error Unsupported architecture
 ^
1 error generated.
make: *** [src/common/pa_cpuload.lo] Error 1
bash-3.2$ 
```

I'm guessing this is related to the `-arch arm64` flag, as my Mac is an Intel-based one.

Hope that helps!

--- a/configure
+++ b/configure
@@ -2539,6 +2539,14 @@ ac_compiler_gnu=$ac_cv_c_compiler_gnu
 
 
 
+PAMAC_TEST_PROGRAM="
+  /* cdefs.h checks for supported architectures. */
+  #include <sys/cdefs.h>
+  int main() {
+      return 0;
+  }
+"
+
 ac_aux_dir=
 for ac_dir in "$srcdir" "$srcdir/.." "$srcdir/../.."; do
   if test -f "$ac_dir/install-sh"; then
@@ -15867,14 +15875,7 @@ case "${host_os}" in
               CFLAGS="$CFLAGS -arch $arch"
               cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
-
-int
-main ()
-{
-return 0;
-  ;
-  return 0;
-}
+$PAMAC_TEST_PROGRAM
 _ACEOF
 if ac_fn_c_try_compile "$LINENO"; then :
 

--- a/configure.in
+++ b/configure.in
@@ -11,6 +11,15 @@ dnl Init autoconf and make sure configure is being called
 dnl from the right directory
 AC_INIT([include/portaudio.h])
 
+dnl This is is for testing compilation on Mac OS
+PAMAC_TEST_PROGRAM="
+  /* cdefs.h checks for supported architectures. */
+  #include <sys/cdefs.h>
+  int main() {
+      return 0;
+  }
+"
+
 dnl Define build, build_cpu, build_vendor, build_os
 AC_CANONICAL_BUILD
 dnl Define host, host_cpu, host_vendor, host_os
@@ -244,13 +253,14 @@ case "${host_os}" in
            esac
 
            dnl Pick which architectures to build for based on what
-           dnl the compiler supports.
+           dnl the compiler and SDK supports.
            mac_arches=""
            for arch in x86_64 arm64
            do
               save_CFLAGS="$CFLAGS"
               CFLAGS="$CFLAGS -arch $arch"
-              AC_TRY_COMPILE([], [return 0;],
+              AC_COMPILE_IFELSE(
+                 [AC_LANG_SOURCE([$PAMAC_TEST_PROGRAM])],
                  [
                     if [[ -z "$mac_arches" ]] ; then
                        mac_arches="-arch $arch"

